zipline/pipeline/term.py

```diff
@@ -368,6 +370,13 @@ def __new__(cls,
             # normalize to a tuple so that inputs is hashable.
             inputs = tuple(inputs)
 
+        if outputs is NotSpecified:
+            outputs = cls.outputs
+        if outputs is NotSpecified:
+            outputs = ()
```

Comment:
I think we should just leave this as `NotSpecified` here, otherwise we can't detect and raise if a user does:

```
class MyDumbFactor(CustomFactor):
    outputs = ()
```